src/main/kotlin/CheckersModel.kt

```diff
@@ -0,0 +1,113 @@
+import kotlin.math.abs
+
+class CheckersModel() : BaseModel(8) {
+    private var whoMoves = Color.WHITE
+
+    override fun canMove(turn: BaseTurn): Pair<Boolean, Any?> {
+        if (turn.playerColor != whoMoves) {
+            //ходит не тот, кто должен бы
+            return Pair(false, null)
+        }
+        //TODO("Проверка выхода за границы")
+        val squareFrom = board[turn.from.first][turn.from.second]
```

Comment:
These constructions may be also simplified if you will create separate `Board` class and overload `[ ]` operator. In such case you may write something like `val (squareFrom, squareTo) = board[turn]` instead of these two lines